cypress/e2e/tests/administration/jira-connection/negative.test.ts

```diff
@@ -0,0 +1,162 @@
+    const now = new Date();
+    now.setDate(now.getDate() + 1);
+    const end = new Date(now.getTime());
+    end.setFullYear(end.getFullYear() + 1);
+    let migrationWave: MigrationWave;
+
+    before("Login and create required credentials", function () {
+        // Perform login
+        login();
+        // Defining and creating credentials to be used in further tests
+        jiraBasicCredential = new JiraCredentials(
+            getJiraCredentialData(CredentialType.jiraBasic, useTestingAccount)
+        );
+
+        jiraBasicCredential.create();
+
+        // Defining Jira connection data with correct credentials
+        jiraCloudConnectionData = getJiraConnectionData(
+            jiraBasicCredential,
+            isSecure,
+            useTestingAccount
+        );
+
+        jiraCloudConnection = new Jira(jiraCloudConnectionData);
+
+        // Defining and creating dummy credentials to be used further in tests
+        jiraBasicCredentialInvalid = new JiraCredentials(
+            getJiraCredentialData(CredentialType.jiraBasic, !useTestingAccount)
+        );
+
+        jiraBasicCredentialInvalid.create();
+
+        // Defining Jira connection data with incorrect credentials
+        jiraCloudConnectionDataIncorrect = getJiraConnectionData(
+            jiraBasicCredentialInvalid,
+            isSecure,
+            useTestingAccount
+        );
+
+        jiraCloudConnectionIncorrect = new Jira(jiraCloudConnectionDataIncorrect);
+
+        applicationList = createMultipleApplications(2);
+    });
+
+    beforeEach("Login", function () {
+        cy.intercept("DELETE", "/hub/migrationwaves*/*").as("deleteWave");
+    });
+
+    it("Creating Jira connection with incorrect credentials", () => {
+        jiraCloudConnectionIncorrect.create();
+        jiraCloudConnectionIncorrect.validateState(expectedToFail);
+    });
+
+    it("Removing Jira connection in use by wave", () => {
+        const issueType = JiraIssueTypes.task;
+        let projectName = "";
+        jiraCloudConnection.create();
+        jiraCloudConnection.validateState(!expectedToFail);
+
+        // Defining and creating new migration wave
+        migrationWave = new MigrationWave(
+            data.getRandomWord(8),
+            now,
+            end,
+            null,
+            null,
+            applicationList
+        );
+        migrationWave.create();
+
+        jiraCloudConnection
+            // Getting all projects on Jira side
+            .getAllProjects()
+            .then((project) => {
+                // Checking that project is not a NULL or undefined
+                expect(!!project[0]).to.eq(true);
+                projectName = project[0].name;
+                return jiraCloudConnection.getIssueType(issueType);
+            })
+            .then((issue) => {
+                // Checking that Jira issue type is not a NULL or undefined
+                expect(!!issue).to.eq(true);
+
+                // Exporting wave to Jira
```

Comment:
**suggestion:** Comments like this one add noise to the code, the function `exportToIssueManager` is explanatory enough

More info: https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#comments-rules

Reply:
Fixed